### PR TITLE
Supporting all ASCII characters in table widget

### LINKED ISSUE #5445: [Bug] Non non-ascii character in column names don't show up correctly in the Table widget
## Description

A user reported that non-ascii characters such as Från (which means From in Swedish) don't show up correctly in Table column headers.

### Steps to reproduce the behaviour:

1. Create a Google sheet with non-ascii characters data
2. Connect it to a Table widget
3.  You will see word Från shown as Fr_n

gsheet_error

### Important Details

- Version: Cloud 
- Browser: Chrome
- Environment: Release

### PR DESCRIPTION
**Pull request by techbhavin**
## Description

> Fixed non-Ascii character `Från` shown as `Fr_n` in Column header

Fixes #5445 

## Type of change

- Bugfix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ]  I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: bug/table-ascii-char 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 51.49 **(-0.01)** | 33.46 **(-0.01)** | 29.3 **(0)** | 52.12 **(0)**
 :green_circle: | app/client/src/components/designSystems/appsmith/TableComponent/TableHelpers.ts | 100 **(0)** | 90 **(10)** | 100 **(0)** | 100 **(0)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 50.59 **(-0.24)** | 39.41 **(-0.84)** | 36.21 **(0)** | 54.77 **(-0.27)**</details>

### CODE REVIEW

**Comment by mohanarpit** [comment]
@techbhavin Is there a Jest test case that can be added here to assert this scenario? Seems like an ideal candidate for unit testing.

**Review by mohanarpit — APPROVED** [review]

File: app/client/src/utils/migrations/TableWidget.ts [hunk 1]
```diff
@@ -168,7 +168,7 @@ export const tableWidgetPropertyPaneMigrations = (
 };
 
 const removeSpecialChars = (value: string, limit?: number) => {
```
Comment (mohanarpit) [inline app/client/src/utils/migrations/TableWidget.ts:170]: Why do we have the exact same function in 3 different files? Can we not just re-use the function in TableHelpers class?

Comment (techbhavin) [reply app/client/src/utils/migrations/TableWidget.ts:170]: 1. In migration/TableWidget we can reuse the function exported from TableHelpers. As it was an old migration I did not do it deliberately.
2. In the TableWidget/derived.js file I don't think we can import a TS file function into the JS file

**Review by mohanarpit — APPROVED** [review]

**Comment by YogeshJayaseelan** [comment]
@somangshu Verified this fix and now ascii characters are showing up correctly

**Note:**
Google sheet is not working in this deploy view and hence i have tested this using mongo plugin instead of google sheet

Screenshot 2021-07-23 at 3 52 29 PM

**Review by somangshu — APPROVED** [review]

**Comment by somangshu** [comment]
Rerunning all tests, Failure in chart_spec.js

**Pull request merged**